### PR TITLE
Implemented integration tests for Metrics API

### PR DESCRIPTION
**Pull request by piosz**
ref https://github.com/kubernetes/kubernetes/issues/23376

as promised in https://github.com/kubernetes/heapster/pull/1161#issuecomment-219452619

### CODE REVIEW

File: metrics/apis/metrics/handlers.go [hunk 1]
```diff
@@ -95,7 +95,11 @@ func (a *Api) nodeMetricsList(request *restful.Request, response *restful.Respon
 
 func (a *Api) nodeMetrics(request *restful.Request, response *restful.Response) {
 	node := request.PathParameter("node-name")
-	response.WriteEntity(a.getNodeMetrics(node))
+	m := a.getNodeMetrics(node)
+	if m == nil {
+		response.WriteError(http.StatusNotFound, fmt.Errorf("No metrics for ode %v", node))
```
Comment (jszczepkowski) [inline metrics/apis/metrics/handlers.go]: Aren't we missing return here?

Comment (piosz) [inline metrics/apis/metrics/handlers.go]: Good catch!

**Comment by jszczepkowski** [comment]
It will be nice to extend the test and check, if all pods/nodes are correctly returned by master API, however, it can be addressed in another PR.

LGTM.

**Comment by piosz** [comment]
Will do. Thanks for the review!

**Pull request merged**